### DISCUSSION TITLE
What is appropriate mode when I disable a submit button until all fields are valid.

### DISCUSSION THREAD

**Discussion by pkino** [react-hook-form/react-hook-form#3357] [Q&A] [answered]
In my case, I disable a submit button until all fields are valid.
If a form has some buttons or checkboxes and the mode is `onBlur`, users have to touch other places than that buttons to fire blur. Therefore, I set the mode `onChange`.
However, `onChange` is deprecated from the point of view of performance.
Are there any options?

**Comment by bluebill1049** [comment, marked as answer]
i think i need CSB to get my head clear on this one.

**Reply by pkino** [reply]
https://codesandbox.io/s/wizardly-cdn-n6cpk?file=/src/App.tsx

CSB is in the case of `onBlur`.
Even if you choose the last radio button, if you push somewhere else of the button, the submit button will not become able.

**Reply by bluebill1049** [reply]
you can do onClick with `trigger` as well.

**Reply by pkino** [reply]
Thank you for answering!

**Reply by bluebill1049** [reply]
np :)
